Samples fail to build for `qemu_xtensa/sample_controller32/mpu` (`BUILD_ASSERT` failure)

**Describe the bug**
Building any sample targetting the `qemu_xtensa/sample_controller32/mpu` platform results in build failure.

**To Reproduce**
Steps to reproduce the behavior:
1. `west build -p -b qemu_xtensa/sample_controller32/mpu samples/hello_world/`
2. See error

**Expected behavior**
The Hello World sample builds successfully.

**Impact**
Showstopper.

**Logs and console output**
```
In file included from /work/zephyr/include/zephyr/toolchain.h:50,
                 from /work/zephyr/include/zephyr/kernel_includes.h:23,
                 from /work/zephyr/include/zephyr/kernel.h:17,
                 from /work/zephyr/arch/xtensa/core/prep_c.c:6:
/work/zephyr/include/zephyr/toolchain/gcc.h:87:36: error: static assertion failed: ""
   87 | #define BUILD_ASSERT(EXPR, MSG...) _Static_assert((EXPR), "" MSG)
      |                                    ^~~~~~~~~~~~~~
/work/zephyr/arch/xtensa/core/prep_c.c:20:1: note: in expansion of macro 'BUILD_ASSERT'
   20 | BUILD_ASSERT(CONFIG_DCACHE_LINE_SIZE == XCHAL_DCACHE_LINESIZE);
      | ^~~~~~~~~~~~
[63/133] Building C object zephyr/arch/arch/xtensa/core/CMakeFiles/arch__xtensa__core.dir/mpu.c.obj
ninja: build stopped: subcommand failed.
FATAL ERROR: command exited with status 1: /usr/bin/cmake --build /work/zephyr/build
```

**Environment (please complete the following information):**

 - OS: Linux
 - Toolchain: Zephyr SDK 0.17.0
 - Commit SHA or Version used: 23232e6bb23fea1a1271ad58cbf58e877ac103ed

--- a/boards/qemu/xtensa/Kconfig.defconfig
+++ b/boards/qemu/xtensa/Kconfig.defconfig
@@ -11,6 +11,7 @@ config IPM_CONSOLE_STACK_SIZE
 
 # Must match XCHAL_DCACHE_LINESIZE form core-isa.h
 config DCACHE_LINE_SIZE
+	default 4 if BOARD_QEMU_XTENSA_SAMPLE_CONTROLLER32_MPU
 	default 32
 
 endif # BOARD_QEMU_XTENSA

